react/src/components/app.js

```diff
@@ -1,29 +1,34 @@
 import React, { Component } from 'react';
-import Mainpage from './Mainpage';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      memes: this.props.data.memes,
-      randMeme: this.props.data.memes[0]
+      categories: [],
     };
-    this.handleClick = this.handleClick.bind(this);
   }
 
-  handleClick() {
-    let random_meme = this.state.memes[Math.floor(Math.random()*this.state.memes.length)];
-    this.setState({ randMeme: random_meme });
+  componentDidMount(){
+    $.ajax({
+        method: "GET",
+        url: "/categories.json",
+      })
+      .done(data => {
```

Comment:
Consider updating your Hound config as indicated (`esversion: 6` in your .yml file.)